src/common/models/config.model.ts

```diff
@@ -170,8 +170,8 @@ export interface IConfigCustomAction extends IConfigMethod {
 }
 
 export interface IConfigPagination {
-  source: 'query' // TODO: add 'headers'
-  type: 'lazy-loading' // TODO: add 'buttons'
+  source: 'query'
+  type: 'lazy-loading' | 'buttons'
   params: IConfigPaginationParams
   fields?: IConfigPaginationFields
```

Comment:
why `fields` are optional?

Reply:
Fields are optional because all properties there are optional. The API does not necessarily return any information about the pagination. The one that could be mandatory is `total` but there could be an API that is not aware of the total amount of items a route has. In that case we assume there can always be a next page.